WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormRemoteActionUseCase.swift

```diff
@@ -241,9 +247,14 @@ private extension ProductFormRemoteActionUseCase {
     func updatePasswordRemotely(product: EditableProductModel,
                                 password: String?,
                                 onCompletion: @escaping (Result<String?, Error>) -> Void) {
-        // Only update product password if available and user is authenticated with WPCom.
+
+        // Update the product password using the `updateSitePostPassword` method only if:
+        // 1) A password is provided.
+        // 2) The user is authenticated with WPCom.
+        // 3) The store is not eligible for the new `password` field introduced in WC 8.1.
+        // Otherwise, update the password locally in the Product model, introduced in WC 8.1.
         guard let updatedPassword = password,
-              stores.isAuthenticatedWithoutWPCom == false else {
+              stores.isAuthenticatedWithoutWPCom == false || !ProductPasswordEligibilityUseCase().isEligibleForWooProductPasswordEndpoint() else {
```

Comment:
The comment suggests that all conditions should be met. Should we replace `||` with `,` or `&&`?

Reply:
The comment here was not very clear, as it is correct to have an OR condition. Therefore, I updated the comment to make it clearer: 3bb6bc4c9331aa45e97bb6790775ee1e6982ba86